File: intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        
        HashSet<Integer> set1 = new HashSet<>();
        HashSet<Integer> set2 = new HashSet<>();

        for (int i : nums1) set1.add(i);
        for (int i : nums2) set2.add(i);

        set1.retainAll(set2);
        
        int[] arr = new int[set1.size()];
        int index = 0;
        for(int num : set1) arr[index++] = num;
        return arr;
        
    }
}

File: intersection-of-two-arrays/intersection-of-two-arrays.rs
impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        use std::collections::HashSet;
            
        nums1.into_iter()
            .collect::<HashSet<i32>>()
            .intersection(&nums2.into_iter().collect::<HashSet<i32>>())
            .into_iter()
            .map(|&x| x)
            .collect()
        
    }
}